Services/Component/classes/class.ilPlugin.php

```diff
@@ -394,23 +394,45 @@ function getTablePrefix()
 	}
 
 	/**
-	 * Update all languages
+	 * Update all or selected languages
+	 * @var array|null	$a_lang_keys	keys of languages to be updated (null for all)
 	 */
-	public function updateLanguages()
+	public function updateLanguages($a_lang_keys = null)
 	{
-		global $ilCtrl;
 		ilGlobalCache::flushAll();
 		include_once("./Services/Language/classes/class.ilObjLanguage.php");
 
+		// get the keys of all installed languages if keys are not provided
+		if(!isset($a_lang_keys))
+		{
+			$a_lang_keys = array();
+			foreach (ilObjLanguage::getInstalledLanguages() as $langObj)
+			{
+				if ($langObj->isInstalled())
```

Comment:
Isn't this a bit redudant, or are there some situations where getInstalledLanguages() return languages, which are not installed?